packages/expo-cli/src/schemes.ts

```diff
@@ -81,6 +91,55 @@ export async function getDevClientSchemeAsync(projectRoot: string): Promise<stri
   return matching;
 }
 
+async function getManagedDevClientSchemeAsync(projectRoot: string): Promise<string> {
+  const { exp, staticConfigPath, dynamicConfigPath } = getConfig(projectRoot, {
+    skipSDKVersionRequirement: true,
+  });
+  if (exp.scheme) {
+    return exp.scheme;
+  }
+
+  Log.warn(
+    '\nDev Client: No URI schemes could be found in configuration, this is required for opening the project\n'
+  );
+  if (dynamicConfigPath && !staticConfigPath) {
+    // Dynamic config (app.config.js/app.config.ts) can't be automatically updated.
+    Log.log(
+      `Add a common scheme in ${dynamicConfigPath} or provide a scheme with the ${Log.chalk.cyan(
+        '--scheme'
+      )} flag\n`
+    );
+    throw new AbortCommandError();
+  }
+
+  const { scheme } = await prompt({
+    type: 'text',
+    name: 'scheme',
+    validate: value =>
+      /^[a-z0-9-]+$/.test(value) || 'Only lowercase characters/numbers (a-z, 0-9) or hyphen (-)',
+    message: `Please choose a URI scheme for your app`,
+    initial: slugify(exp.slug, {
+      // Remove non-allowed characters and additionally hyphens, which are allowed but not idiomatic.
+      remove: /[^a-z0-9]/,
+      // Convert to lower case.
+      lower: true,
+    }),
+  });
+  if (staticConfigPath) {
+    // if app.json exists we can update it
+    const config = JsonFile.read(staticConfigPath, { json5: true });
+    config.expo = { ...(config.expo as object), scheme };
+    await JsonFile.writeAsync(staticConfigPath, config);
+    Log.log(`Scheme '${scheme}' updated in ${path.basename(staticConfigPath)}`);
+  } else {
```

Comment:
this would just ignore the dynamic config and create a potentially unused file

Reply:
Existing dynamic config is handled by an earlier if – see https://github.com/expo/expo-cli/pull/3523/files#diff-ef67bc980d05133cc7b87b450cf4a04e1546cae27683bc12a749a925bb79cf09R105

If we land in this else branch it means there's no config (dynamic or static) at all so we just create an app.json.